src/Core/Services/Fido2AuthenticatorService.cs

```diff
@@ -0,0 +1,233 @@
+                );
+
+                throw new NotAllowedError();
+            }
+            
+            try {
+                var selectedFido2Credential = selectedCipher.Login.MainFido2Credential;
+                var selectedCredentialId = selectedFido2Credential.CredentialId;
+
+                if (selectedFido2Credential.CounterValue != 0) {
+                    ++selectedFido2Credential.CounterValue;
+                }
+
+                await _cipherService.UpdateLastUsedDateAsync(selectedCipher.Id);
+                var encrypted = await _cipherService.EncryptAsync(selectedCipher);
+                await _cipherService.SaveWithServerAsync(encrypted);
+
+                var authenticatorData = await GenerateAuthData(
+                    rpId: selectedFido2Credential.RpId,
+                    userPresence: true,
+                    userVerification: userVerified,
+                    counter: selectedFido2Credential.CounterValue
+                );
+
+                var signature = await GenerateSignature(
+                    authData: authenticatorData,
+                    clientDataHash: assertionParams.Hash,
+                    privateKey: selectedFido2Credential.KeyBytes
+                );
+
+                return new Fido2AuthenticatorGetAssertionResult
+                {
+                    SelectedCredential = new Fido2AuthenticatorGetAssertionSelectedCredential
+                    {
+                        Id = GuidToRawFormat(selectedCredentialId),
+                        UserHandle = selectedFido2Credential.UserHandleValue
+                    },
+                    AuthenticatorData = authenticatorData,
+                    Signature = signature
+                };
+            } catch {
+                _logService.Info(
+                    "[Fido2Authenticator] Aborting because no matching credentials were found in the vault."
+                );
+
+                throw new UnknownError();
+            }
+        }
+
+    private async Task<List<CipherView>> FindCredentialsById(PublicKeyCredentialDescriptor[] credentials, string rpId)
+    {
+        var ids = new List<string>();
+
+        foreach (var credential in credentials)
+        {
+            try
+            {
+                ids.Add(GuidToStandardFormat(credential.Id));
+            }
+            catch {}
+        }
+
+        if (ids.Count == 0)
+        {
+            return new List<CipherView>();
+        }
+
+        var ciphers = await _cipherService.GetAllDecryptedAsync();
+        return ciphers.FindAll((cipher) =>
+            !cipher.IsDeleted &&
+            cipher.Type == CipherType.Login &&
+            cipher.Login.HasFido2Credentials &&
+            cipher.Login.MainFido2Credential.RpId == rpId &&
+            ids.Contains(cipher.Login.MainFido2Credential.CredentialId)
+        );
+    }
+
+    private async Task<List<CipherView>> FindCredentialsByRp(string rpId)
```

Comment:
⛏️ Add async suffix:
```suggestion
    private async Task<List<CipherView>> FindCredentialsByRpAsync(string rpId)
```

Reply:
Fixed